filter.go

```diff
@@ -25,15 +34,28 @@ type Config struct {
 
 	// Backlog of how many packets we are happy to buffer in memory
 	Backlog int
+
+	// If non-zero, uses ipv4.PacketConn.ReadBatch, using the size of the batch given.
+	// Defaults to 1 on Darwin/FreeBSD and 8 on Linux.
+	BatchSize int
 }
 
 // NewPacketFilter creates a packet filter object wrapping the given packet
 // connection.
 func NewPacketFilter(conn net.PacketConn) *PacketFilter {
+	// This is derived from quic codebase.
+	var batchSize = 0
+	switch runtime.GOOS {
+	case "linux":
+		batchSize = 8
+	case "freebsd", "darwin":
+		batchSize = 1
+	}
```

Comment:
In theory this is not needed, seems go does the right thing anyway.

Reply:
Should we then better rely on what go does so we stay up to date when changes happen?

Reply:
This avoids using ReadBatch on Windows, which is not supported anyway.